docs/DeveloperGuide.md

```diff
@@ -1,34 +1,670 @@
+display the full week. Otherwise, it displays the current day’s schedule in a list form, with the `Task` items sorted 
+by date.
+
+![Figure 4.8-1](images/DeveloperGuide/Figure 4.8-1.png)
+_Figure 4.8-1: Sequence diagram for displaying week or day schedule_
+
+##### Implementation
+
+The timetable is structured in such a way that the first column always starts with the schedule of the current day, 
+followed by that of the next 6 days. This is so that the user always sees 7 days ahead, rather than a typical fixed 
+format _(e.g. from Monday to Sunday)_.
+
+Changes to the timetable are updated at every call of the `showTimetable()` method, which first involves the generation 
+of the timetable by loading the contents of the `TaskList` into it, then printing it row by row.
+
+The timetable is modelled using an `ArrayList`, with each entry containing a row of the timetable as a string. The 
+individual cell entries of the timetable are collected by iterating through each day, each time slot and then through 
+the `TaskList` to see which tasks fall on that particular day and are held during that particular time slot.
+
+The cell entries which fall on the same time slot and hence the same row, are collected into an array and formatted into 
+a string, before it is finally added to the `ArrayList` of timetable rows.
+
+##### Design Considerations
+
+To ensure that the displayed timetable is easy to read and offers a quick view of the user’s schedule, especially that 
+of the current day, the timetable is not made to be fixed. The display schedule commands must thus iterate through the 
+entire `TaskList` every time it is called, in order to arrange the `Tasks` accordingly and update any changes.
+
+Because of the way the timetable time slots increment on an hourly basis, functions were implemented to ensure the 
+timings of `Tasks` were rounded to the hour. This was an intentional design choice to keep the timetable neat and not 
+overloaded with too much details.
+
+### 4.9 Displaying Free Time and Sleep Time (Daniel)
+
+##### Implementation
+
+The `freeTime` command displays to the user their longest block of free time for that day. The `sleepTime`
+ command displays to the user how much time they have available to sleep based on that day’s and the
+ next day’s schedule. Both commands are implemented similarly. They both find the longest uninterrupted block
+ of free time within a certain time period by checking if individual hour-long time blocks in this time period
+ are free. The commands then use the start and end time values found to calculate a duration, and pass all
+ three values to the Ui to display to the user. Figure 4.9-1 shows the sequence diagram for the `freeTimeCommand`,
+ and Figure 4.9-2 shows the sequence diagram for the `sleepTimeCommand`.
+
+![Figure 4.9-1](images/DeveloperGuide/Figure%204.9-1.png)  
+_Figure 4.9-1: Sequence diagram for freeTimeCommand execution_
+
+
+![Figure 4.9-2](images/DeveloperGuide/Figure%204.9-2.png)  
```

Comment:
Should this be a dotted line?
![image](https://user-images.githubusercontent.com/60306071/97669529-bdb6e600-1a9d-11eb-9d40-3df5b8f2e324.png)